packages/cashier/src/pages/payment-agent/payment-agent-card/paymen-agent-card-deposit-details.jsx

```diff
@@ -0,0 +1,131 @@
+import PropTypes from 'prop-types';
+import classNames from 'classnames';
+import { toJS } from 'mobx';
+import React from 'react';
+import { Icon, Text, Money } from '@deriv/components';
+import { localize } from '@deriv/translations';
+
+export const Detail = ({ action, children, className, has_red_color, icon, title }) => {
+    const detail = Array.isArray(children) ? children : [children];
+    return (
+        <div
+            className={classNames('payment-agent-card__detail', {
+                [className]: !!className,
+            })}
+        >
+            {icon && (
+                <div className='payment-agent-card__detail-icon-container'>
+                    <Icon icon={`IcCashier${icon}`} />
+                </div>
+            )}
+            <div>
+                {detail.map((child, id) => (
+                    <React.Fragment key={id}>
+                        {title && (
+                            <Text as='p' line_height='s' size='xs'>
+                                {title}
+                            </Text>
+                        )}
+                        {action || !title ? (
+                            <Text
+                                as='a'
+                                color={has_red_color ? 'red' : 'prominent'}
+                                href={`${action ? `${action}:` : ''}${child}`}
+                                line_height='s'
+                                size={!title ? 'xxs' : 'xs'}
+                                weight='bold'
+                                className='payment-agent-card__detail-link'
+                            >
+                                {child}
+                                {id === detail.length - 1 ? '' : ', '}
+                            </Text>
+                        ) : (
+                            <Text as='p' line_height='s' size='xs' weight='bold'>
+                                {child}
+                            </Text>
+                        )}
+                    </React.Fragment>
+                ))}
+            </div>
+        </div>
+    );
+};
+
+Detail.propTypes = {
+    action: PropTypes.string,
+    children: PropTypes.oneOfType([PropTypes.array, PropTypes.element, PropTypes.string]),
+    className: PropTypes.string,
+    has_red_color: PropTypes.bool,
+    icon: PropTypes.string,
+    title: PropTypes.string,
+};
+
```

Comment:
@george-usynin-binary Not sure why we have two components in the same file 🤔 Is it the coding style we follow here?
I would suggest separating them and making smaller components as generic as possible so we can reuse them in other places.

Reply:
sometimes we can use two components in the same file because they are unique, but here i agree with you, because we also use Detail component in other file, will refactor

Reply:
sorted